website/blog/2023-02-14-pgo.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Optimizing Databend Binary Builds with Profile-guided Optimization
+description: rustc supports doing profile-guided optimization (PGO). Let's see how we can make it work for Databend.
+slug: profile-guided-optimization
+date: 2023-02-13
+tags: [databend, Profile-guided Optimization]
+authors:
+- name: PsiACE
+  url: https://github.com/PsiACE
+  image_url: https://github.com/PsiACE.png
+---
+
+Previously, a voice in the community suggested that we try profile-guided optimization ([#9387](https://github.com/datafuselabs/databend/issues/9387)). Let's see how we can use Rust to build a PGO-optimised Databend!
+
+## Background
+
+Profile-guided Optimization is a compiler optimisation technique, which collects typical execution data (possible branches) during program execution and then optimises for inlining, conditional branches, machine code layout, register allocation, etc.
+
+The background to the introduction of this technique is that static analysis techniques only consider code performance improvements without actually executing the program. However, these optimisations may not be fully effective. In the absence of runtime information, the compiler cannot take into account the actual execution of the program.
+
+PGO allows data to be collected based on application scenarios in a production environment, allowing the optimiser to optimise speed for hot code paths and size for cold code paths, producing faster and smaller code for applications.
```

Comment:
PGO allows data to be collected based on application scenarios in a production environment, so the optimizer can optimize the speed for hot code paths and size for cold code paths and produce faster and smaller code for applications.